### PR TITLE
Replace uploaded activities

### PR DESCRIPTION
**Pull request by merefield**
Allow a user to replace an existing upload from the Edit Topic meta area.

### CODE REVIEW

File: assets/javascripts/discourse/templates/components/activity-file-edit.hbs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{{#if siteSettings.zspace_display_upload_below_topic_title}}
+  {{#if buffered.zspace_upload.url}}
+    {{d-button href=buffered.zspace_upload.url icon="download" translatedLabel=buffered.zspace_upload.original_filename}}
+    <span class='col'>
+      {{d-button action=(action "removeUpload") actionParam=upload class="second btn-default btn-default cancel-edit" icon="times"}}
+    </span>
+  {{/if}}
+    <label class="btn" disabled={{uploading}} title="{{i18n "zspace.upload_a_file"}}">
```
Comment (angusmcleod) [inline assets/javascripts/discourse/templates/components/activity-file-edit.hbs]: @merefield Is there not a generic upload button template we can use (that just takes a title and label as properties)?

File: plugin.rb [hunk 2]
```diff
@@ -27,4 +30,22 @@
...
   }
-end
+
+  ::PostRevisor.track_topic_field(:zspace_upload) do |tc, zspace_upload|
+      old_zspace_upload = tc.topic.custom_fields['zspace_upload']
+      if tc.record_change('zspace_upload', old_zspace_upload, zspace_upload)
+        begin
+          old_upload_id = JSON.parse(old_zspace_upload)["id"]
+          Upload.find_by(id: old_upload_id).destroy_all
+        rescue => e
```
Comment (angusmcleod) [inline plugin.rb]: @merefield We should only be rescuing JSON::ParserError here. See further: https://stackoverflow.com/questions/10048173/why-is-it-bad-style-to-rescue-exception-e-in-ruby

File: assets/javascripts/discourse/components/activity-file-edit.js.es6 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      bootbox.alert(I18n.t("zspace.upload_successful"));
+    }
+  },
+ 
+
+  actions: {
+    removeUpload(upload) {
+
+    var that = this;
+    return bootbox.confirm(
```
[the remark this answers is not in the document]
Comment (merefield) [reply assets/javascripts/discourse/components/activity-file-edit.js.es6]: Good point, yes.  I can amend the text to say after hitting the tick?  I still think there should be a warning which won't be possible without interfering with the tick action.

File: assets/javascripts/discourse/templates/components/activity-file-edit.hbs (hunk 1 above)
Comment (merefield) [reply assets/javascripts/discourse/templates/components/activity-file-edit.hbs]: You mean an upload component?  I'll take a look, but this gives us more control.

File: plugin.rb (hunk 2 above)
Comment (merefield) [reply plugin.rb]: But what if the find_by returns a Nil Class?

Comment (angusmcleod) [reply plugin.rb]: You can tweak the query so it won't cause an error when there are no records:
```
Upload.where(id: old_upload_id).destroy_all
```

**Pull request merged**